### PR TITLE
nessus mapping sorting

### PR DESCRIPTION
**Pull request by bigdevwhale**

### CODE REVIEW

File: protected/controllers/NessusmappingController.php [hunk 1]
```diff
@@ -378,12 +378,41 @@ public function actionFilterVulns() {
...
+
+                case NessusMapping::FILTER_SORT_CHECK:
+                    $sortBy = "checks.name";
+                    break;
+            }
+
+            $criteria = new CDbCriteria();
+            $criteria->addCondition("nessus_mapping_id = :mapping_id");
+            $criteria->params = ["mapping_id" => $form->mappingId];
+            $criteria->addInCondition('nessus_host', $form->hosts);
```
Comment (anton-belousov) [inline protected/controllers/NessusmappingController.php:411]: Double quotes here and below.

File: protected/controllers/NessusmappingController.php [hunk 2]
```diff
@@ -378,12 +378,41 @@ public function actionFilterVulns() {
...
+            $criteria->addInCondition('nessus_rating', $form->ratings);
+            $criteria->join = 'LEFT JOIN checks ON checks.id = check_id';
+            $criteria->order = $sortBy." ".$sortDirection;
```
Comment (anton-belousov) [inline protected/controllers/NessusmappingController.php:414]: - `$sortBy . " " . $sortDirection` -- see code style guideline
- sortBy or sortDirection may be null here, making the SQL incorrect - see form validation

File: protected/forms/NessusMappingVulnFilterForm.php [hunk 3]
```diff
@@ -20,11 +20,26 @@ class NessusMappingVulnFilterForm extends FormModel  {
...
+     */
+    public $sortDirection;
+
+    /**
      * Nessus mapping vulns form rules
      * @return array
      */
     public function rules() {
         return [
+            [["sortBy", "sortDirection"],'safe'],
```
Comment (anton-belousov) [inline protected/forms/NessusMappingVulnFilterForm.php:42]: They are not safe - add arrays of allowed values here. Also don't forget spaces after the comma

File: protected/models/NessusMapping.php [hunk 4]
```diff
@@ -10,6 +10,15 @@
  * @property NessusMappingVuln $vulns
  */
 class NessusMapping extends ActiveRecord {
+
```
Comment (anton-belousov) [inline protected/models/NessusMapping.php:13]: Empty line is not needed here.

File: protected/models/NessusMapping.php [hunk 5]
```diff
@@ -10,6 +10,15 @@
...
+    // sorting
+    const FILTER_SORT_ISSUE = 1;
+    const FILTER_SORT_RATING = 2;
+    const FILTER_SORT_CHECK = 3;
+
+    // sorting direction
+    const FILTER_SORT_ASCENDING = 1;
+    const FILTER_SORT_DESCENDING = 2;
     /**
```
Comment (anton-belousov) [inline protected/models/NessusMapping.php:22]: But here we need an empty line.

**Review by anton-belousov — CHANGES_REQUESTED** [review]
See code review notes.

File: protected/controllers/NessusmappingController.php (hunk 2 above)
Comment (bigdevwhale) [reply protected/controllers/NessusmappingController.php:414]: sortBy and sortDirection can not be null, if you see that error just hard reload page

**Comment by anton-belousov** [comment]
Please create a new PR from swissfondue repo.

**Pull request closed without merging**